packages/x-data-grid/src/hooks/features/rows/useGridRows.ts

```diff
@@ -234,11 +237,27 @@ export const useGridRows = (
         previousCache: apiRef.current.caches.rows,
       });
 
-      throttledRowsChange({ cache, throttle: true });
+      throttledRowsChange({ cache, throttle });
     },
     [props.signature, props.getRowId, throttledRowsChange, apiRef],
   );
 
+  const setLoading = React.useCallback<GridRowApi['setLoading']>(
+    (loading) => {
+      if (loading === props.loading) {
+        return;
+      }
+      logger.debug(`Setting loading to ${loading}`);
+      apiRef.current.setState((state) => ({
+        ...state,
+        rows: { ...state.rows, loading },
+      }));
+      apiRef.current.caches.rows.loadingPropBeforePartialUpdates = loading;
+      apiRef.current.forceUpdate();
```

Comment:
We should avoid adding `forceUpdate` calls, we're supposed to have all the tools we need to re-render what needs to re-render without this.